src/ebonite/api/images.py

```diff
@@ -1,80 +1,176 @@
+        """
+        Creates image in repository and optionally builds
+        ---
+        parameters:
+          - name: body
+            in: body
+            required: true
+            schema:
+              required:
+                - name
+                - buildable
+              properties:
+                name:
+                  type: string
+                buildable:
+                  schema:
+                    properties:
+                      object_type:
+                        type: string
+                        description: model or pipeline
+                      object_id:
+                        type: integer
+                        description: id of buildable object
+                builder_args:
+                  description: dictionary with arguments for image build
+          - name: skip_build
+            in: query
+            type: integer
+            required: false
+            default: 0
+        responses:
+          201:
+            description: Image built successfully
+          400:
+            description: Image with given name already exist
+          404:
+            description: Either buildable with given id does not exist or there was an error with using
+                         any of provided builder_args
+        """
+        skip_build = False if not request.args.get('skip_build') else bool(int(request.args.get('skip_build')))
+        body = request.get_json(force=True)
+        BuildableValidator(**body['buildable'])
+        buildable = pj.deserialize(body.pop('buildable'), Buildable)
+        builder_args = None
+        builder_args = body.pop('builder_args', builder_args)
         try:
-            image = ebonite.build_image(name=image.name, model=model)
-            return jsonify(pj.dumps(image)), 201
+            image = ebonite.create_image(buildable, name=body['name'], builder_args=builder_args, skip_build=skip_build)
+            return jsonify(pj.serialize(image)), 201
         except ExistingImageError:
-            return jsonify({'errormsg': f'Image with name {image.name} already exists'}), 400
+            return jsonify({'errormsg': f'Image with name {body["name"]} already exists'}), 400
+        except (ValueError, TypeError, NonExistingPipelineError, NonExistingModelError) as e:
+            return jsonify({'errormsg': str(e)}), 404
 
-    @blueprint.route('/<int:id>', methods=['PATCH'])
-    def update_image(id: int):
-        body = request.get_json(force=True)
-        body['id'] = id
-        image = UpdateImageBody.from_data(body)
-        try:
-            ebonite.meta_repo.update_image(image)
-            return jsonify({}), 204
-        except NonExistingImageError:
-            return jsonify({'errormsg': f'Project with id {id} does not exist'}), 404
+    # @blueprint.route('/<int:id>', methods=['PATCH'])
+    # def update_image(id: int):
+        # TODO: It's bad. Maybe recreate image when trying to update meta image?
```

Comment:
no, we can update meta before we even build the first version. so lets keep the api as close to client as possible